numpy/lib/function_base.py

```diff
@@ -1482,42 +1482,46 @@ def angle(z, deg=False):
     return a
 
 
-def _unwrap_dispatcher(p, discont=None, axis=None):
+def _unwrap_dispatcher(p, discont=None, axis=None, *, interval_size=2*pi):
     return (p,)
 
 
 @array_function_dispatch(_unwrap_dispatcher)
-def unwrap(p, discont=pi, axis=-1):
+def unwrap(p, discont=None, axis=-1, *, interval_size=2*pi):
     """
-    Unwrap by changing deltas between values to 2*pi complement.
-
-    Unwrap radian phase `p` by changing absolute jumps greater than
-    `discont` to their 2*pi complement along the given axis.
-
+    Unwrap by changing deltas between values to complement.
+    
+    For the default case where `interval_size= 2*pi`, `discont=pi`,
+    It unwraps radian phase `p` by changing absolute jumps greater 
+    than `discont` to their 2*pi complement along the given axis. Jumps equal 
+    to `discont` are not changed.
+    
+    In general it unwrapps a signal `p` by changing absolute jumps 
+    greater than `discont` to their `interval_size` complementary values.
+    
     Parameters
     ----------
     p : array_like
         Input array.
     discont : float, optional
-        Maximum discontinuity between values, default is ``pi``.
+        Maximum discontinuity between values, default is ``interval_size/2``.
     axis : int, optional
         Axis along which unwrap will operate, default is the last axis.
-
+    interval_size: float, optional
```

Comment:
Thoughts on using the shorter name `period`?

Reply:
Yeah, sounds also good! I just picked the first name which came to my mind! Shall I rename it?

Reply:
Yeah, I think so - we use `period` already in `np.interp`, and as far as I can tell the meaning is the same.